minijinja/src/environment.rs

```diff
@@ -206,11 +206,30 @@ impl<'source> Environment<'source> {
     pub fn render_str<S: Serialize>(&self, source: &str, ctx: S) -> Result<String, Error> {
         // reduce total amount of code faling under mono morphization into
         // this function, and share the rest in _eval.
-        self._render_str(source, Value::from_serializable(&ctx))
+        self._render_str("<string>", source, Value::from_serializable(&ctx))
     }
 
-    fn _render_str(&self, source: &str, root: Value) -> Result<String, Error> {
-        let name = "<string>";
+    /// Like [`render_str()`](Self::render_str()), but provide a name for the
+    /// template to be used instead of the default `<string>`.
+    ///
+    /// ```
+    /// # use minijinja::{Environment, context};
+    /// let env = Environment::new();
+    /// let rv = env.render_str_named("Hello {{ name }}", "template", context! { name => "World" });
+    /// println!("{}", rv.unwrap());
+    /// ```
+    pub fn render_str_named<S: Serialize>(
```

Comment:
I wonder if the template name shouldn't go first. That aligns with `add_template`.

Reply:
Yeah, I teeter-tottered between the two options. The reason I chose this was because the method is named `render_str_named`, so I wanted to have the `str` and then the `name`. But it could easily be argued that after "named" should come the name: `render_str_named("foo", "{{ the template }}")`. I don't have any preference.

Reply:
Maybe `render_named_str` is a better name anyways. Would probably pick that.

Reply:
`render_named_str(name, template, txt)` or `render_named_str(template, name, ctxt)`?

Reply:
`render_named_str(name, template, ctxt)` analog to `add_template(name, template)?.render(ctxt)`